### PR TITLE
add less concepts, 1-d CA sequencer

### PR DESCRIPTION
**Pull request by dndrks**
- utilizes @markwheeler 's `Passersby` engine
- grid-enabled, but not grid-dependent
- midi-enabled, two channels of notes
- [here](https://llllllll.co/uploads/default/original/3X/d/1/d19e15fb63831f7883d2660f77c83cc56811a6bc.png) for grid legend

### CODE REVIEW

File: scripts/dndrks/less_concepts.lua [hunk 1]
```diff
@@ -0,0 +1,803 @@
...
+local ch_1 = 1
+local ch_2 = 1
+local semi = 0
+local presets = {}
+local preset_count = 0
+local active_notes_v1 = {}
+local active_notes_v2 = {}
+
+beatclock = require 'beatclock'
+clk = beatclock.new()
```
Comment (artfwo) [inline scripts/dndrks/less_concepts.lua]: `clk` and `beatclock` could be local too

File: scripts/dndrks/less_concepts.lua [hunk 2]
```diff
@@ -0,0 +1,803 @@
...
+clk_midi = midi.connect()
+clk_midi.event = clk.process_midi
+
+engine.name = "Passersby"
+passersby = require "mark_eats/passersby"
+
+-- this section is all maths + computational events
+
+-- maths: translate the seed integer to binary
+function seed_to_binary()
```
Comment (artfwo) [inline scripts/dndrks/less_concepts.lua]: could also be local, also check functions below

File: scripts/dndrks/less_concepts.lua [hunk 3]
```diff
@@ -0,0 +1,803 @@
...
+    out1 = (rule_as_binary[4] << 7) & 128
+  elseif compare (seed_pack1, neighborhoods6) then
+    out1 = (rule_as_binary[3] << 7) & 128
+  elseif compare (seed_pack1, neighborhoods7) then
+    out1 = (rule_as_binary[2] << 7) & 128
+  elseif compare (seed_pack1, neighborhoods8) then
+    out1 = (rule_as_binary[1] << 7) & 128
+  else out1 = (0 << 7) & 128
+  end
+end
```
Comment (artfwo) [inline scripts/dndrks/less_concepts.lua]: looks like indentation is wrong in this function and below?

File: scripts/dndrks/less_concepts.lua [hunk 4]
```diff
@@ -0,0 +1,803 @@
...
+
+com1()
+com2()
+com3()
+com4()
+com5()
+com6()
+com7()
+com8()
```
Comment (artfwo) [inline scripts/dndrks/less_concepts.lua]: can these generators be generalised as `com(seed_packN, lshift, mask)`?

File: scripts/dndrks/less_concepts.lua [hunk 5]
```diff
@@ -0,0 +1,803 @@
...
+
+-- allow user to define the transposition of voice 1 and voice 2, simultaneous changes to MIDI and Passersby engine
+function transpose(semitone)
+  semi = semitone
+end
+
+-- everything that happens when the script is first loaded
+function init()
+  math.randomseed(os.time())
+  math.random(); math.random(); math.random()
```
Comment (artfwo) [inline scripts/dndrks/less_concepts.lua]: why generate 3 random numbers that aren't used?

File: scripts/dndrks/less_concepts.lua (hunk 1 above)
Comment (dndrks) [reply scripts/dndrks/less_concepts.lua]: done in update!

File: scripts/dndrks/less_concepts.lua (hunk 2 above)
Comment (dndrks) [reply scripts/dndrks/less_concepts.lua]: done in update!

File: scripts/dndrks/less_concepts.lua (hunk 4 above)
Comment (dndrks) [reply scripts/dndrks/less_concepts.lua]: super good note, thank you for the help!

File: scripts/dndrks/less_concepts.lua (hunk 3 above)
Comment (dndrks) [reply scripts/dndrks/less_concepts.lua]: resolved in update!

File: scripts/dndrks/less_concepts.lua (hunk 5 above)
Comment (dndrks) [reply scripts/dndrks/less_concepts.lua]: based on comments from folks in Lua forums, the first 'random' number is not pure random -- it was advised to trigger a few at the start, to overcome. is this true?

**Comment by dndrks** [comment]
@artfwo : new version posted! 
61deca4
thank you so much for the excellent feedback, that really helped clean things up! also, fixed some grid bugs with nil snapshot recall :)

**Pull request merged**